src/Semaphore/Mutex/Redis.php

```diff
@@ -44,40 +44,67 @@ public function testAndSetLock(): bool
                 // If the client that registered for the mutex resource ID is connected, the mutex is held by another client.
                 if ($mutexClientIsConnected === false) {
                     // If not, try to obtain the lock by registering on the mutex resource ID.
-                    $this->_getRedisClient()->multi();
-                    $this->_getRedisClient()->set($key, $processUUID);
-                    $reply = $this->_getRedisClient()->exec();
-
-                    // If the mutex resource ID was not set by another client, the mutex is obtained by this client.
-                    if ($reply[0] === true) {
+                    if ($this->setLockKey($key, $processUUID)) {
                         $this->_hasLock = true;
                     }
                 }
-            }else {
+            } else {
                 // If the mutex resource ID is not set, try to obtain the mutex.
-                $this->_getRedisClient()->multi();
-                $this->_getRedisClient()->set($key, $processUUID);
-                $reply = $this->_getRedisClient()->exec();
-                if (is_array($reply) && $reply[0] === true) {
+                if ($this->setLockKey($key, $processUUID)) {
                     $this->_hasLock = true;
-                }elseif ($reply !== false) {
-                    $type = gettype($reply);
-                    throw new \UnexpectedValueException("Reply is of type [$type]");
                 }
             }
-        }else {
+        } else {
             throw new \LogicException('The mutex already has obtained a lock.');
         }
 
         return $this->_hasLock;
     }
 
+    protected function setLockKey(string $key, string $processUUID) : bool
+    {
+        $this->_getRedisClient()->multi();
+        $this->_getRedisClient()->set($key, $processUUID);
+
+        $reply = $this->_getRedisClient()->exec();
+
+        if (is_array($reply) && $reply[0] === true) {
+            return true;
+        }
+
+        if ($reply === false) {
+            return false;
+        }
+
+        $type = gettype($reply);
+        throw new \UnexpectedValueException("Reply is of type [$type]");
+    }
+
+    public function testLock() : bool
+    {
+        $keyValue = $this->_getRedisClient()->get($this->_getKey());
+
+        // not likely if nothing cleans up the key after the lock is released
```

Comment:
```suggestion
        // not likely since the entire key is deleted normally when the lock is released
```

**non-blocker**: I think this would be more accurate. In our current implementation, if we found an empty string, wouldn't that mean our `uuid` generation method has likely failed. 

This could also throw an Exception since it's not expected for this value to be empty.

Reply:
The `empty()` check is for `false` (i.e. there's no value at that key), not an empty string

Reply:
It couldn't even be an empty string currently, the `_getKey()` implementation starts with a hard-coded `'/'`

Reply:
I'll change it to an explicit `=== false` check though, since that's clearer